sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -66,7 +64,8 @@ internal HttpMessage CreateUploadRequest(string ruleId, string streamName, Reque
             {
                 request.Headers.Add("Content-Encoding", contentEncoding);
             }
-            if (contentEncoding == "gzip")
+            // if contentEncoding == "gzip" that means content is already gzipped, so we shouldn't gzip again
+            if (contentEncoding != "gzip")
```

Comment:
Should this just be a null check? If some other encoding is specified I assume we would still want to avoid gzipping.